wordpress-mobile/WordPress-iOS
# Blaze: switch to using canBlaze to determine Blaze eligibility

Part of #20915
WPKit PR: https://github.com/wordpress-mobile/WordPressKit-iOS/pull/609

## Description
- Adds a new readonly  `canBlaze` property for the Blog model that determines blaze eligiblity based on the following:
    - The `can_blaze` option returned from `/me/sites` is true 
    - The user is an admin for the site
- Removes the `isBlazeApproved` property from the Blog model

## How to test

<details><summary>isAdmin</summary>
<p>

1. Switch to a site that's eligible for Blaze (See peeHDf-zm-p2 for more info on Blaze requirements)
2. ✅ The Blaze dashboard card is displayed
3. ✅ The Blaze menu item is displayed
4. Switch to a site where you're not an admin
5. ✅ The Blaze dashboard card is NOT displayed
6. ✅ The Blaze menu item is NOT displayed

</p>
</details> 

<details><summary>Site visibility</summary>
<p>

1. Switch to a site that's eligible for Blaze (See peeHDf-zm-p2 for more info on Blaze requirements)
2. ✅ The Blaze dashboard card is displayed
3. ✅ The Blaze menu item is displayed
4. Go to Site Settings > Privacy, and set the blog to private
5. ✅ The Blaze dashboard card is NOT displayed
6. ✅ The Blaze menu item is NOT displayed
7. Go to Site Settings > Privacy, and set the blog to public
8. ✅ The Blaze dashboard card is displayed
9. ✅ The Blaze menu item is displayed

</p>
</details> 

<details><summary>Posts list screen</summary>
<p>

1. Switch to a site that's eligible for Blaze (See peeHDf-zm-p2 for more info on Blaze requirements)
2. Create three types of posts: public, private, and password protected
9. Go to the posts lists screen and select the published tab
10. On the public post, tap the ellipsis button
11. ✅ A "Promote with Blaze" alert action is displayed
12. On the private post, tap the ellipsis button
13. ✅ A "Promote with Blaze" alert action is NOT displayed
14. On the password protected post, tap the ellipsis button
15. ✅ A "Promote with Blaze" alert action is NOT displayed
16. On the posts list screen, switch to the drafts tab
17. On any post, tap the ellipsis button
18. ✅ A "Promote with Blaze" alert action is NOT displayed
19. Switch to a site that's not eligible for Blaze
20. On any post, tap the ellipsis button
21. ✅ A "Promote with Blaze" alert action is NOT displayed

</p>
</details> 

<details><summary>Pages list screen</summary>
<p>

1. Switch to a site that's eligible for Blaze (See peeHDf-zm-p2 for more info on Blaze requirements)
2. Create three types of pages: public, private, and password protected
3. Go to the pages lists screen and select the published tab
4. On the public page, tap the ellipsis button
6. ✅ A "Promote with Blaze" alert action is displayed
10. On the private page, tap the ellipsis button
11. ✅ A "Promote with Blaze" alert action is NOT displayed
12. On the password protected page, tap the ellipsis button
13. ✅ A "Promote with Blaze" alert action is NOT displayed
14. On the pages list screen, switch to the drafts tab
15. On any page, tap the ellipsis button
16. ✅ A "Promote with Blaze" alert action is NOT displayed
17. Switch to a site that's not eligible for Blaze
18. On any page, tap the ellipsis button
19. ✅ A "Promote with Blaze" alert action is NOT displayed

</p>
</details> 


## Regression Notes
1. Potential unintended areas of impact
Blaze eligibilty

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Followed the above testing steps

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
